ophyd/v2/core.py

```diff
@@ -188,14 +190,23 @@ async def connect_children(device: Device, prefix: str, sim: bool):
         async def connect(self, prefix: str = "", sim=False):
             await connect_children(self, prefix + self.prefix, sim)
     """
-    coros = {
-        k: c.connect(prefix, sim)
-        for k, c in device.__dict__.items()
-        if k != "parent" and isinstance(c, Device)
-    }
+
+    coros = {}
+    for name, child_device in get_device_children(device):
+        if isinstance(child_device, DeviceDict):
+            coros.update(child_device.nested_connect(prefix, sim))
+        else:
+            coros.update({name: child_device.connect(prefix, sim)})
```

Comment:
We call this when implementing the `connect` method of a Device, so this will never be called with a `DeviceDict`, so we can go back to the simpler implementation:

```suggestion
    coros = {
        name: child_device.connect(prefix, sim) 
        for name, child_device in get_device_children(device)
    }
```